test/socket.js

```diff
@@ -96,4 +95,14 @@ describe('socket', function(){
       socket.compress(false).emit('hi');
     });
   });
+
+  it('should store query string as a property', function(done){
+    var socket = io('/abc', {query: {a:'b'}}); //passes in as a query obj
+    var socket2 = io('/abcd?b=c&d=e'); //passes in as a query string
+    expect(socket.query).to.be('a=b');
+    expect(socket2.query).to.be('b=c&d=e');
```

Comment:
The corresponding PR on [socket.io](https://github.com/socketio/socket.io/pull/2422) has the test that checks for the query parameter being sent over. Do you think that's ok?